### PR TITLE
add database viewer

### PR DESCRIPTION
**Pull request by pwjablonski**
Hey @jtessler!

This adds a database viewer to the menu bar. It's a basic implementation based on trivia app. 

Note it uses a new version of Material UI since the current version is deprecated. Will have to go through and update old component new version of Material UI.

### CODE REVIEW

**Comment by pwjablonski** [comment]
There is still of a bit of code to clean up so not ready to merge but let me know if you think this is the right direction!

**Comment by pwjablonski** [comment]
Also is there any issue around exposing the user keys publicly?

File: src/Header.js [hunk 1]
```diff
@@ -61,6 +61,13 @@ class Header extends Component {
           </Link>
           <Subheader>Active Games</Subheader>
           <MenuItem primaryText="Not implemented yet" disabled={true} />
+          <Link style={{textDecoration: 'none'}} to="/dataViewer">
```
Comment (jtessler) [inline src/Header.js]: Add a `<Subheader>` to separate this menu item from "Active Games", e.g. "Debug Tools".

File: src/Header.js [hunk 2]
```diff
@@ -61,6 +61,13 @@ class Header extends Component {
...
+            <MenuItem
+                onClick={() => this.setDrawerOpen(false)}
+                primaryText="View Game Data"
+                // disabled={this.isInWaitingRoom()}
```
Comment (jtessler) [inline src/Header.js]: Remove commented out code

File: src/DataViewer.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      <Paper className="Container">  
+        <ExpansionPanel
+            style={{boxShadow: 'none'}}
+            disabled={this.props.disabled}
+            onChange={(e, expanded) => this.handleExpanded(expanded)}>
+
+            <ExpansionPanelSummary
+                style={{paddingLeft: 0}}
+                expandIcon={<ExpandMoreIcon />}>
+            <Typography>
```
Comment (jtessler) [inline src/DataViewer.js:78]: Nit: indent the `<Typography>` tag by 2 spaces

File: src/DataViewer.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                Expand to see JSON view of entire database
+            </Typography>
+            </ExpansionPanelSummary>
+
+            <ExpansionPanelDetails
+                style={{display: 'block', paddingLeft: 0, textAlign: 'left'}}>
+            {this.renderDatabaseViewer()}
```
Comment (jtessler) [inline src/DataViewer.js:85]: Nit: indent by 2 spaces

File: src/DataViewer.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    if (expanded) {
+      switch (this.state.db_state) {
+        case DB_STATE.UNINITIALIZED:
+          this.setState({ db_state: DB_STATE.LOADING });
+          firebase.database().ref("/").once("value", (snapshot) => {
+            // Check if user data firebase download is done yet.
+            var db_state = this.state.user_data === null
+                ? DB_STATE.LOADING : DB_STATE.LOADED;
+            this.setState({
+              db_state: db_state,
```
Comment (jtessler) [inline src/DataViewer.js]: Why wouldn't this **always** be `db_state: DB_STATE.LOADED`? The "check if user data firebase download is done yet" seems irrelevant -- if you enter this callback, then the firebase download must be done.

**Review by jtessler — COMMENTED** [review]
Looks great, Peter! Couldn't have done it better myself. Only have minor comments.

**Comment by jtessler** [comment]
> Will have to go through and update old component new version of Material UI.

Good call. I filed a new [issue](https://github.com/jtessler/studio-multiplayer-game/issues/16) to track this migration.

> Also is there any issue around exposing the user keys publicly?

No, there isn't. I filed a new [issue](https://github.com/jtessler/studio-multiplayer-game/issues/15) to expose them.

**Comment by jtessler** [comment]
I [started migrating game code](https://github.com/jtessler/studio-multiplayer-game/commit/677635878b72239a5cae77562bbb654d0ecbae67) to use the new Material UI as you suggested (so students get a good reference if they choose to use it too). This obviously conflicts with your pull request now -- sorry!

**Comment by pwjablonski** [comment]
@jtessler I think this should be good to go now. 

Quick indenting question - is everything 2 spaces except for component props which is 4?

File: src/firebaseConfig.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const firebaseConfig = {
```
Comment (pwjablonski) [inline src/firebaseConfig.js]: Woops left this in!

File: src/DataViewer.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import React, { Component } from 'react';
+import ReactJson from 'react-json-view'
+import Typography from '@material-ui/core/Typography';
+import firebase from 'firebase';
+import Paper from 'material-ui/Paper';
```
Comment (jtessler) [inline src/DataViewer.js]: Can you find an alternative to using the deprecated `<Paper>` Material UI element? I just removed them from the other components.

**Comment by jtessler** [comment]
> Quick indenting question - is everything 2 spaces except for component props which is 4?

I use 4 spaces for component props that are wrapped to a new line and 2 spaces for childen tags, e.g.:

```
<Component
    with={true}
    lots="of"
    props={false}>
  <ChildComponent />
<Component>
```

File: src/DataViewer.js (hunk 7 above)
Comment (pwjablonski) [reply src/DataViewer.js]: Yup!
